### ISSUE TITLE
Failing test nerdctl restart

### ISSUE THREAD

**Issue by wzshiming** [kubernetes-sigs/kwok#259]
### Which jobs are failing?

https://github.com/kubernetes-sigs/kwok/actions/runs/4104392465/jobs/7092806797

### Which tests are failing?

nerdctl restart

### Since when has it been failing?

#249

**Comment by wzshiming** [comment]
It looks like there are still a lot of problems with nerdctl's stop, and if it's not fixed we'll need to revert #249.

**Comment by zwpaper** [comment]
yes, I also found it is related to nerdctl, and did not find an args to fix it, I will dig into nerdctl later this day.

did you have time to do the revert? or should I raise a PR to revert and rebase https://github.com/kubernetes-sigs/kwok/pull/244 onto it?

**Comment by zwpaper** [comment]
I found it is easy to create a revert PR when walking around 🤣, created one https://github.com/kubernetes-sigs/kwok/pull/260

**Comment by wzshiming** [comment]
> yes, I also found it is related to nerdctl, and did not find an args to fix it, I will dig into nerdctl later this day.

Yeah, it is better to find the reason.

**Comment by zwpaper** [comment]
I found it looks like `compose stop` does not really stop the containers, containers will appear after a short time disappear.

but it is not yet confirmed, I will dig into nerdctl to confirm this.

**Comment by wzshiming** [comment]
Fixes by #260

/close

**Comment by wzshiming** [comment]
/reopen

The old nerdctl test was also very unstable

**Comment by nikola-jokic** [comment]
This might have been fixed with https://github.com/kubernetes-sigs/kwok/blob/910014238bfda07409a1f92b50ac161305243f07/test/kwokctl/kwokctl_restart_test.sh#LL57C9-L57C9

The issue was that one scheduler container if I remember correctly did not get a chance to start up with nerdctl, the delay was longer than set in tests

**Comment by wzshiming** [comment]
Thank you, that's great.

/close
